starfish/codebook.py

```diff
@@ -371,52 +460,27 @@ def decode_euclidean(self, intensities: IntensityTable) -> IntensityTable:
             qualities
 
         """
+        # normalize both the intensities and the codebook
+        norm_intensities, norms = self._normalize_features(intensities, norm=norm)
```

Comment:
from an offline convo w/ @kevinyamauchi -- expose norm as a tuple? if it's (None, None) then don't normalize. If it's (2, 1) then you can normalize features and codebook w/ L2 and L1 norm respectively -- even thought that would be weird.

Reply:
I think this deserves a bit more thought, I don't find that API very intuitive.